ts/features/wallet/creditCard/screen/CreditCardDetailScreen.tsx

```diff
@@ -45,19 +52,53 @@ const CreditCardDetailScreen: React.FunctionComponent<Props> = props => {
   }, [storeCreditCard, setWalletExisted]);
 
   if (storeCreditCard !== undefined) {
+    const paymentCardData = pipe(
+      storeCreditCard,
+      ({ info }) =>
+        sequenceS(O.Monad)({
+          // all or nothing, if one of these is missing we don't show the card
```

Comment:
Are we sure about this?

Reply:
unless we want to add a series of mock values, there is no datum that can be undefined if we want to avoid the user seeing a bunch of "NaN" and "undefined"s.
In any case if any of these is missing a skeleton will be rendered so the user will not see any void or weird UI.
(also, in the previous screen this was not handled, possibly rendering weird values, so my guess is that it's just an error in the DTOs which marks mandatory parameters as optional)

Reply:
If something is marked as optional in the DTO we should treat as optional. I think it's better to show an empty space or a fallback string (like a `-`) instead of not displaying the card at all. What do you think?

Reply:
@mastro993 after a discussion with @CrisTofani it appears that it truly is a DTO problem, and expecting the PM to give correctly formed data was the mantra back in the day those DTOs were born.
I say we keep it like this, with a skeleton fallback and when the new wallet rolls out we will surely have to come back at it, as with all of the remaining wallet screens.